test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -672,8 +607,383 @@ TEST_CASE("progress notification", "[sync][session][progress]") {
             callback_was_called = false;
             current_transferred = original_transferrable + 250;
             current_transferrable = 1228;
-            progress.update(current_transferred, current_transferrable, 199, 591, 1, 1);
+            progress.update(current_transferred, current_transferrable, 199, 591, 1);
             CHECK(!callback_was_called);
         }
     }
 }
+
+#if REALM_ENABLE_AUTH_TESTS
+
+struct TestSetup {
+    TableRef get_table(const SharedRealm& r)
+    {
+        return r->read_group().get_table("class_" + table_name);
+    }
+
+    size_t add_objects(SharedRealm& r, int num = 5)
+    {
+        CppContext ctx(r);
+        for (int i = 0; i < num; ++i) {
+            // use specifically separate transactions for a bit of history
+            r->begin_transaction();
+            Object::create(ctx, r, StringData(table_name), std::any(make_one(i)));
+            r->commit_transaction();
+        }
+        return get_table(r)->size();
+    }
+
+    virtual SyncTestFile make_config() = 0;
+    virtual AnyDict make_one(int64_t idx) = 0;
+
+    std::string table_name;
+};
+
+struct PBS : TestSetup {
+    PBS()
+    {
+        table_name = "Dog";
+    }
+
+    SyncTestFile make_config() override
+    {
+        const auto schema = get_default_schema();
+        return SyncTestFile(session.app(), partition, schema);
+    }
+
+    AnyDict make_one(int64_t /* idx */) override
+    {
+        return AnyDict{{"_id", std::any(ObjectId::gen())},
+                       {"breed", std::string("bulldog")},
+                       {"name", random_string(1024 * 1024)}};
+    }
+
+    TestAppSession session;
+    const std::string partition = random_string(100);
+};
+
+struct FLX : TestSetup {
+    FLX(const std::string& app_id = "flx_sync_progress")
+        : harness(app_id)
+    {
+        table_name = (*harness.schema().begin()).name;
+        harness.do_with_new_user([](std::shared_ptr<SyncUser>&&) {});
```

Comment:
I don't think this is needed. But if you want an user, `create_user_and_log_in` would suffice.